subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/transform/ArtifactTransformBuildOperationIntegrationTest.groovy

```diff
@@ -0,0 +1,609 @@
+                artifactName == "producer.jar"
+                dependenciesConfigurationIdentity == null
+            }
+            transformType == "MakeRed"
+            sourceAttributes == [color: "blue", artifactType: "jar"]
+            fromAttributes == [color: "blue"]
+            toAttributes == [color: "red"]
+        }
+
+        def executeTransformationOp2 = executeTransformationOps[1]
+        with(executeTransformationOp2.details) {
+            transformerName == "MakeGreen"
+            subjectName == "producer.jar (project :producer)"
+            with(transformationIdentity) {
+                nodeType == "ARTIFACT_TRANSFORM"
+                buildPath == ":"
+                projectPath == ":consumer"
+                targetVariant.componentId == [buildPath: ":", projectPath: ":producer"]
+                targetVariant.attributes == [color: "green", artifactType: "jar"]
+                targetVariant.capabilities == [[group: "colored", name: "producer", version: "unspecified"]]
+                artifactName == "producer.jar"
+                dependenciesConfigurationIdentity == [buildPath: ":", projectPath: ":consumer", name: "resolver"]
+            }
+            transformType == "MakeGreen"
+            sourceAttributes == [color: "red", artifactType: "jar"]
+            fromAttributes == [color: "red"]
+            toAttributes == [color: "green"]
+        }
+    }
+
+    def checkExecutionPlanMatchingDependencies(List<PlannedNode> plannedNodes, List<NodeMatcher> nodeMatchers) {
+        Map<NodeId, List<String>> expectedDependencyPhantomIdsByNodeId = [:]
+        Map<NodeId, String> phantomIdByNodeId = [:]
+
+        for (def plannedNode : plannedNodes) {
+            def matchers = nodeMatchers.findAll { it.matchNode(plannedNode) }
+            assert matchers.size() == 1, "Expected exactly one matcher for node ${plannedNode.nodeIdentity}, but found ${matchers.size()}"
+            def nodeMatcher = matchers[0]
+            def nodeId = getNodeId(plannedNode.nodeIdentity)
+            phantomIdByNodeId[nodeId] = nodeMatcher.phantomId
+            expectedDependencyPhantomIdsByNodeId[nodeId] = nodeMatcher.dependencyPhantomIds
+        }
+
+        for (def plannedNode : plannedNodes) {
+            def nodeId = getNodeId(plannedNode.nodeIdentity)
+            def expectedDependencyPhantomIds = expectedDependencyPhantomIdsByNodeId[nodeId]
+            assert expectedDependencyPhantomIds != null, "No expected dependencies for node $plannedNode"
+
+            def actualDependencyPhantomIds = plannedNode.nodeDependencies.collect { depIdentity ->
+                def depNodeId = getNodeId(depIdentity)
+                def depPhantomId = phantomIdByNodeId[depNodeId]
+                assert depPhantomId != null, "No phantom id for node $depIdentity"
+                depPhantomId
+            }
+
+            assert actualDependencyPhantomIds.toSet() == expectedDependencyPhantomIds.toSet()
```

Comment:
```suggestion
            assert actualDependencyPhantomIds ==~ expectedDependencyPhantomIds
```

💄 You can use Spock's `==~` comparison: https://spockframework.org/spock/docs/2.2/utilities.html#collection-conditions

Reply:
Done